fix no-coalescing fuzzerbugs

jonsneyers opened the pull request:
~~Couldn't find any fuzzerbugs running it locally for a short while, but still good to test both coalesced and non-coalesced decoding to find potential bugs.~~

Fixes a fuzzer-found bug in the uint8 buffer case (where stride was not computed correctly in the no-coalescing case).
Also adds an ASSERT in dec_external_image to hopefully find more such bugs and to turn buffer overruns into an abort.

veluca93 commented on lib/jxl/dec_external_image.cc:
```diff
@@ -315,12 +315,12 @@ Status ConvertChannelsToExternal(const ImageF* channels[], size_t num_channels,
                        " vs %" PRIuS,
                        stride, bytes_per_pixel * xsize);
   }
+  JXL_ASSERT(out_callback || out_size >= ysize * stride);
```
Not sure how I feel about adding an assert here, shouldn't we return a failure instead?

jonsneyers replied:
Makes sense. Changed the condition to correspond to https://github.com/libjxl/libjxl/issues/994 (don't require padding after last row), and returning failure instead of having an assert.

veluca93 commented:
Can we split adding fuzzing and fixing a fuzzbug in two PRs or, at least, two commits?

jonsneyers commented:
> Can we split adding fuzzing and fixing a fuzzbug in two PRs or, at least, two commits?

Yes. Updated this one to just fix the bug. I'll make a new one to add fuzzing.
